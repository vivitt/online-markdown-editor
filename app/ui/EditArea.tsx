"use client";
import { useEffect, useState } from "react";

type Props = {
  onChange: (event: React.ChangeEvent<HTMLTextAreaElement>) => void;
};

export default function EditArea({ onChange }: Props) {
  const [value, setValue] = useState("");

  useEffect(() => {
    const saved = window.localStorage.getItem("markdown-content");
    if (saved) {
      setValue(saved);
    }
  }, []);

  return (
    <div className="flex px-3 font-chivo w-full">
      <div
        id="line-numbers"
        aria-hidden="true"
        className="w-5 grid pr-3 text-slate-400"
      >
        {value.split("\n").length > 0 &&
          value.split("\n").map((el, index) => {
            return (
              <span className="h-6" key={index + 1}>
                {index + 1}
              </span>
            );
          })}
      </div>

      <textarea
        id="edit"
        name="edit"
        aria-autocomplete="list"
        data-language="markdown"
        className="resize-none w-full h-screen focus:outline-0"
        onChange={(e) => {
          onChange(e);
          setValue(e.target.value);
        }}
        value={value}
        placeholder="Add your content here..."
      ></textarea>
    </div>
  );
}
