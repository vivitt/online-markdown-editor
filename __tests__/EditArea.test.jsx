import { cleanup, render, screen } from "@testing-library/react";
import { afterEach, describe, it, expect, spyOn, vi } from "vitest";
import userEvent from "@testing-library/user-event";
import EditArea from "../app/ui/EditArea";

const getItemSpy = vi.spyOn(Storage.prototype, "getItem");
const setItemSpy = vi.spyOn(Storage.prototype, "setItem");
const user = userEvent.setup();
const LOCAL_STORAGE_KEY = "markdown-content";

describe("editor buttons", () => {
  afterEach(cleanup);
  afterEach(() => {
    localStorage.clear();
    getItemSpy.mockClear();
    setItemSpy.mockClear();
  });

  it.skip("image button adds ![add alt text](paste image link)", () => {
    const { container } = render(<EditArea />);
    const imgButton = screen.getAllByRole("button", { name: "image" });
  });
});
