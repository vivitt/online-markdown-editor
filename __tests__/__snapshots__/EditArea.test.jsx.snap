// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`editor buttons > image button adds ![add alt text](paste image link) 1`] = `
<div>
  <div>
    <button
      aria-pressed="true"
    >
      Save
    </button>
    <button>
      New
    </button>
    <button>
      Create a README file
    </button>
  </div>
  <div
    class="mt-2 border-t-2"
  >
    <div
      class="flex"
    >
      <div
        class="w-full border"
      >
        <h2
          class="p-2 m-2 text-slate-600 text-xl border bg-white rounded-sm inline-flex"
        >
          Edit
        </h2>
        <div
          class="flex"
          data-testid="edit-panel"
        >
          <div
            class="flex px-3 font-chivo w-full"
          >
            <div
              aria-hidden="true"
              class="w-5 grid pr-3 text-slate-400"
              id="line-numbers"
            >
              <span
                class="h-6"
              >
                1
              </span>
            </div>
            <textarea
              aria-autocomplete="list"
              class="resize-none w-full h-screen focus:outline-0"
              data-language="markdown"
              id="edit"
              name="edit"
              placeholder="Add your content here..."
            />
          </div>
        </div>
      </div>
      <div
        class="w-full border"
      >
        <h2
          class="p-2 m-2 text-slate-600 text-xl border bg-white rounded-sm inline-flex"
        >
          Preview
        </h2>
        <div
          class="flex"
          data-testid="preview-panel"
        >
          <div
            class="prose px-3 w-full bg-white h-screen"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
